sessions/04-using-delay-distributions-to-model-the-data-generating-process-of-an-epidemic/index.qmd

```diff
@@ -2,3 +2,291 @@
+It needs to be run inside the course repository so that the `here()` commands below find the stan model files.
+:::
+
+# Simulating observations from a time series of infections
+
+As above we first simulate the process that generates the data, before using the same model to conduct inference.
+
+## Delay distributions and convolutions
+
+In the last session we simulated _individual outcomes_ from a delay distribution, and then re-estimated the corresponding parameters.
+However, sometimes we do not have data on these individual-level outcomes, either because they are not recorded or because they cannot be shared, for example due to privacy concerns.
+At the population level, individual level delays translate into _convolutions_.
+If we have a time series of infections $I_t$ ($t=1, 2, 3, \ldots, t_\mathrm{max}$), where $t$ denotes the day on which the infections occur, and observable outcomes occur with a delay given by a delay distribution $p_i$ ($i=0, 1, 2, \dots, p_\mathrm{max}$), where $i$ is the number of days after infection that the observation happens, then the number of observable outcomes $C_t$ on day $t$ is given by
+
+$$
+C_t = \sum_{i=0}^{i=p_\mathrm{max}} I_{t-i} p_i
+$$
+
+In words, to get the number of observable outcomes on day $t$ is given by the sum of infections on all previous days multiplied by the probability that those infections are observed on day $t$.
+For example, the observable outcomes $C_t$ could be the number of symptom onsets on day $t$ and $p_i$ is the incubation period.
+
+We can use exactly the same data as in the previous session but this time aggregate it into a time series:
+
+```{r aggregate}
+data(infection_times)
+## assume a start date, e.g. 10 May of this year
+start_date <- ymd("2024-05-10")
+df <- infection_times |>
+  mutate(
+    incubation_period = rgamma(n(), shape = 5, rate = 1),
+    infection_date = floor_date(start_date + infection_time, unit = "day"),
+    onset_date = floor_date(
+      start_date + infection_time + incubation_period,
+      unit = "day"
+    ),
+  )
+## infection time series
+inf_time_series <- df |>
+  count(infection_date)
+head(inf_time_series)
+```
+
+The resulting time series does contain any days with no infections.
+We can make our calculations easier by adding them:
+
+```{r missing_to_zero}
+all_dates <- expand(
+  df, infection_date = seq(min(infection_date), max(infection_date), by = "day")
+)
+inf_time_series <- all_dates |>
+  full_join(inf_time_series, by = join_by(infection_date)) |>
+  replace_na(list(n = 0))
+head(inf_time_series)
+```
+
+Now we can convolve the time series with a delay distribution of outcomes to get a time series of outcomes as suggested above.
+However, if we want to use the gamma distribution with shape 5 and rate 1 as before we face a problem.
+The gamma distribution is a _continuous_ distribution but days are _discrete_ entities.
+One solution to this problem is to first _discretise_ the gamma distribution.
```

Comment:
why are we introducing this like they haven't see the censoring session?

Reply:
phrasing is gone now.